### DISCUSSION TITLE
Frigate port 5000 Security Authentication

### DISCUSSION THREAD

**Discussion by Sarelseekat** [blakeblackshear/frigate#1580] [Ask A Question]
Hello

Is there any way to secure access to frigate on port 5000 via a user and password? I'm using the frigate as an I-frame on another instance of Home Assistant. To secure port 5000, I create a firewall rule that blocks all Source IP addresses apart from the IP of the remote instance with the I-Frame. This works well on a PC and Mobile phone app, but the camera feed doesn't work when you leave the site and the mobile phone uses its LTE connection and not the local WIFI.

The only way to make it work at the moment is to provide an unrestricted access to port 5000.

Is there a better way to do it?

**Comment by mickeynygaard** [comment]
Super interested in this as well - I just love the Frigate smooth fast interface...would love to have Home Assistant send me notifications upon motion and then be able to go directly to the Frigate webapp on my phone...but right now I don't see this as an option without opening port 5000 to the entire internet as my phone doesn't have a static IP...

Suggestions anyone?

**Reply by blakeblackshear** [reply]
You can run frigate behind the reverse proxy with auth of your choice. I have used oauth2_proxy and traefik, but there are many different options.

**Reply by Sarelseekat** [reply]
Thanks Blake. That sounds like a practical straight forward way to do it. Will give it a try.

**Reply by farnots** [reply]
@blakeblackshear Do you have in mind other solution that an oauth2_proxy, the github repository is archive so it's in end of live. I tried to find some basic auth or login/password account but I couldn't find any. Any tips ?

**Reply by blakeblackshear** [reply]
@farnots its not EOL: https://github.com/oauth2-proxy/oauth2-proxy

**Reply by farnots** [reply]
@blakeblackshear Oh ok sorry I was on [the forked repositoy](https://github.com/bitly/oauth2_proxy). But did you still know some stuff that could work without having to use a third-party client ? Like a basic login/password proxy in docker ? I couldn't find anything good to secure frigate expose on the web.

**Reply by blakeblackshear** [reply]
Here are more options:

- https://docs.linuxserver.io/general/swag
- nginx reverse proxy manager

**Reply by farnots** [reply]
> Here are more options:
> 
>     * https://docs.linuxserver.io/general/swag
> 
>     * nginx reverse proxy manager

Awesome didn't know `swag`. Thanks a lot for sharing

**Comment by Sarelseekat** [comment]
Unfortunately, my Ignorance is doesn't allow for an immediate grasp of the concept. Traefik seems to explain things a little better, will still have to figure out how I'm going to integrate it with Hass OS. Would have still loved to have native User and Pass Authentication on Frigate:5000. That way there isn't another third party plugin to monitor and maintain.

Hopefully, there will be a native integration for Frigate. We are using Frigate for our neighbourhood watch and it's absolutely fantastic in what it allows us to do over the web with street view CCTV at different houses.

**Reply by blakeblackshear** [reply]
If you already use HassOS, why not use the addon and enable the sidebar? HassOS already has user authentication built in.

**Reply by Pinewold** [reply]
As a tech person, this is what VPN’s were created for, user authentication that gives a local network address in return with full control over what ports are forwarded. 

You can setup the VPN to start automatically

**Reply by rich0** [reply]
Obviously there are a gazillion ways to go about securing it, but application-level authentication is pretty common all the same.  Frigate is actually the only web-based application I'm running that I need to build an additional authentication barrier for.  Right now I'm just using basic auth on traefik though that isn't very friendly with password managers.  I'm looking to get keycloak working with it at some point.

**Comment by fokcuk** [comment]
sorry for resurrecting this. I have it running in HA and its fine. But...
1. Does not use SSL
2. I can just access it on port 5000 without HA and shows all of my cameras 
How do I restrict it to just access via HA?

**Reply by NickM-27** [reply]
You can disable port 5000 and just use ingress in HA sidebar. Otherwise there's plenty of reverse proxy options listed above
